### DISCUSSION TITLE
Pumps not operating until restart

### DISCUSSION THREAD

**Discussion by DrJeff** [tagyoureit/nodejs-poolController#548] [Q&A]
Not sure what happens to stop comms on pumps but I get this all the time the pumps pool, spa, or jets just stop communicating.
here is a replay of me trying to toggle pumps on and off with no actual pumps on or off. 
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/9143563/replay.zip)
And here is a replay of me `pm2 restart` the pool controller and pumps working.
[replay (1).zip](https://github.com/tagyoureit/nodejs-poolController/files/9143578/replay.1.zip)

Any thoughts of why?

**Comment by tagyoureit** [comment]
> [7/19/2022, 10:54:05 AM] warn: Inactivity timeout for 0 serial port /dev/ttyS0 after 10 seconds
[7/19/2022, 10:54:05 AM] info: Serial Port 0 has been closed 0: 
[7/19/2022, 10:54:05 AM] info: Successfully closed 0 serial port /dev/ttyS0
[7/19/2022, 10:54:05 AM] info: Serial port: /dev/ttyS0 request to open successful 9600b 8-none-1

This is in the very beginning of your first set of logs which means the problem happened sometime before.  Your second replay the comms are working just fine.  I'd suggest that you keep the replay capture working from the time it starts comms successfully to the point where it stops.  Then upload the replay and we can see if pinpoint it.

I've also witnessed a problem a while ago with changing the ports that can orphan the pumps and other equipment.  https://github.com/tagyoureit/nodejs-poolController/issues/387.  I wonder if this is related?

**Comment by DrJeff** [comment]
Ok here it is at some point this morning or last night pump comms lost on pool pump and spa jets, but spa pump is working?
The schedule pump did not actually run but the circuit is active. Even manual toggle of circuit doesn't turn it on.
The replay is too big here is a link to file. 
[https://share.wooding5.com/!nVvxcwYHmE/replay%20(2).zip](https://share.wooding5.com/!nVvxcwYHmE/replay%20(2).zip
)

**Reply by tagyoureit** [reply]
Will take a look.  I started to poke around this yesterday.  Comms are like a house of cards, but I did see some items that could be tweaked.

**Reply by DrJeff** [reply]
No biggie thanks whenever is fine. I'll just script restarts if needed for now.

**Reply by tagyoureit** [reply]
23:02:43 seems to be the last comms from the chlor.  Anything happening at that time?  I don't see anything special in the logs.  

<img width="1410" alt="2022-07-20_15-17-13" src="https://user-images.githubusercontent.com/1646840/180091961-8abbaa2e-ae6a-4083-9566-b684a82702d6.png">

**Reply by DrJeff** [reply]
Maybe I should test removing chlorinator from system. I don't see anything special at that point. I could just leave the chlorinator on max and take it out of the comms loop and report back.

**Reply by rstrouse** [reply]
From your logs what is happening here is that your chlorinator stops responding to requests, then in a frantic attempt to get back control Nixie tries 7 times to get it to respond.  Each time it does this it eats just over a second.  Perhaps we are being a bit to persistent and don't really need to ask 7 times but what is happening is that the pump in the meantime is the comms watchdog is looking for responses on the wire and there is radio silence for 10 seconds so Nixie closes the port and re-opens it.   Increase your timeout to 20 seconds for the port and it won't restart the port.  If you have a spare RS485 dongle laying around you cold also assign it to another port if you want to see when it is misbehaving.

It is odd that when it is responding the max time for the response is a <100ms.  Then it just seems to stop answering our requests for a little bit.  Then it will start talking back.  Like it went on vacation or it is rebooting.

**Reply by DrJeff** [reply]
It looks like the removal of the chlorinator solved it so yes your theory seems correct. Now I will keep it off for a day and then I will try with it on and the 20sec delay on the port.
How am I supposed to read the second rs485?

**Reply by rstrouse** [reply]
If you look at the Controller => RS485 Port tab in the config.  You can add another port.  Then in the Chlorinator config set the port to this new port.

**Reply by DrJeff** [reply]
Wow didn't notice the Chlorinator Nixie selection, nice add. I've set it up so long ago it didn't exist before hahahahahaha. It seems to run fine for now not connected but will add it to a secondary port Sunday and test from there on the capture I'll be running for a while did you want me to just have chlorinator included to cut down on size of capture or it's better to tick more boxes in logging?

**Reply by DrJeff** [reply]
Does the Mega-BAS have 2 RS485? Or just 1 in and pass through? Oh it looks like the 8 relay has an RS485 can I access them (I have 2, 8 relay hats)? I don't see the ports in REM.

**Reply by rstrouse** [reply]
Unfortunately, you cannot.  These are all wired to the same UART serial lines.
